src/contrib/mlir/backend/pass/affine_lowerer.cpp

```diff
@@ -1271,10 +1271,22 @@ namespace
         NGRAPH_CHECK(lhs.getType().isa<MemRefType>());
         Type elemTy = lhs.getType().dyn_cast<MemRefType>().getElementType();
 
+        auto origOperand = op->getOperands()[0];
+        auto origIntType =
+            origOperand.getType().cast<NGTensorType>().getElementType().dyn_cast<IntegerType>();
+
         AffineLoopNestBuilder(ivs, lbs, ubs, steps)([&] {
             Value val = iLHS(ivs);
             Value zero = createZeroConstant(elemTy);
-            iRes(ivs) = std_select(val > zero, val, zero);
+            auto cmp = val > zero;
```

Comment:
This overloaded '>' operator is actually generating code so you may want to put it in an 'else' branch after 1287 so that we don't generate two compare ops for unsigned.

Reply:
Done.